src/parsers/espresso/7.2/parser.js

```diff
@@ -0,0 +1,247 @@
+    [alpha, beta, gamma] = [alpha, beta, gamma].map((x) =>
+        x === undefined ? x : (x * 180) / math.PI,
+    );
+    return [alpha, beta, gamma];
+}
+
+/**
+ * @summary Returns cell config from ibrav and celldm(i) parameters
+ *
+ * QE docs: https://www.quantum-espresso.org/Doc/INPUT_PW.html#ibrav
+ * "If ibrav /= 0, specify EITHER [ celldm(1)-celldm(6) ]
+ *   OR [ A, B, C, cosAB, cosAC, cosBC ]
+ *   but NOT both. The lattice parameter "alat" is set to
+ *   alat = celldm(1) (in a.u.) or alat = A (in Angstrom);"
+ *
+ * @param {Object} system - The system parameters from &SYSTEM namelist
+ * @param {Number} system.ibrav - ibrav parameter
+ * @param {Number[]} [system.celldm] - celldm parameters
+ * @param {Number} [system.a] - A parameter in angstroms
+ * @param {Number} [system.b] - B parameter in angstroms
+ * @param {Number} [system.c] - C parameter in angstroms
+ * @param {Number} [system.cosab] - cosAB parameter
+ * @param {Number} [system.cosac] - cosAC parameter
+ * @param {Number} [system.cosbc] - cosBC parameter
+ * @returns {{vectors: Number[][], type: String}}
+ */
+function ibravToCellConfig(system) {
+    const { ibrav, celldm, a, b, c, cosab, cosac, cosbc } = system;
+    if (celldm && a) {
+        throw new Error("Both celldm and A are given");
+    } else if (!celldm && !a) {
+        throw new Error("Missing celldm(1)");
+    }
+
+    if (celldm) celldm.unshift(null); // Bump indices by 1 to make it easier to understand the logic. In QE input file celdm(1) array starts with 1, but parsed starting with 0.
+
+    const type = ibravToCellType(ibrav);
+    const [_a, _b, _c] = getCellConstants(celldm, a, b, c);
+    const [alpha, beta, gamma] = getCellAngles(celldm, cosbc, cosac, cosab);
+
+    const lattice = new Lattice({
+        a: _a,
+        b: _b,
+        c: _c,
+        alpha,
+        beta,
+        gamma,
+        type,
+    });
+    const vectors = primitiveCell(lattice);
+    return { vectors, type };
+}
+
+/**
+ * Parses QE .in file into a Material config object.
+ * @param {String} fileContent - contents of the .in file
+ * @return {{cell: Object, elements: Object[], coordinates: Object[], constraints: Object[], units: String, name: String}}
+ */
+function fromEspressoFormat(fileContent) {
+    const data = parseFortranFile(fileContent);
+
+    if (data.system === undefined) throw new Error("No &SYSTEM section found in input data.");
+    if (data.system.ibrav === undefined) throw new Error("ibrav is required in &SYSTEM.");
+
+    let cell = {};
+    if (data.system.ibrav === 0) {
+        cell = getCellConfig(data.cards);
+        cell.type = Lattice.typeFromVectors(cell.vectors); // not implemented yet, defaults to TRI
+    } else {
+        cell = ibravToCellConfig(data.system);
+    }
+
+    const { elements, coordinates, constraints, units } = getAtomicPositions(data.cards);
+    const name = data.control.title;
+    return { cell, elements, coordinates, constraints, units, name };
```

Comment:
Let's define a class with these properties and have its specific implementation for QE